File: android/app/src/main/kotlin/com/xpx/vault/billing/BillingBootstrap.kt
package com.xpx.vault.billing

import android.app.Application
import com.revenuecat.purchases.LogLevel
import com.revenuecat.purchases.Purchases
import com.revenuecat.purchases.PurchasesConfiguration
import com.xpx.vault.BuildConfig

/**
 * RevenueCat 初始化；[BuildConfig.REVENUECAT_API_KEY] 为空时跳过（例如 CI 未注入 `local.properties`）。
 */
object BillingBootstrap {
    @Volatile
    var isConfigured: Boolean = false
        private set

    fun init(application: Application) {
        val key = BuildConfig.REVENUECAT_API_KEY
        if (key.isBlank()) {
            return
        }
        Purchases.logLevel = if (BuildConfig.DEV_TOOLS) LogLevel.DEBUG else LogLevel.ERROR
        Purchases.configure(
            PurchasesConfiguration.Builder(application, key).build(),
        )
        isConfigured = true
    }
}

File: ios/LumaNox/Core/Billing/BillingBootstrap.swift
import Foundation
import RevenueCat

/// RevenueCat 初始化；未配置 API Key 时跳过（与 Android [BillingBootstrap] 一致）。
enum BillingBootstrap {
    private(set) static var isConfigured = false

    static func configure() {
        guard !isConfigured else { return }
        guard let key = apiKey, !key.isEmpty else { return }
        #if DEBUG
        Purchases.logLevel = .debug
        #else
        Purchases.logLevel = .error
        #endif
        Purchases.configure(withAPIKey: key)
        isConfigured = true
    }

    private static var apiKey: String? {
        if let env = ProcessInfo.processInfo.environment["REVENUECAT_API_KEY"]?
            .trimmingCharacters(in: .whitespacesAndNewlines),
           !env.isEmpty {
            return env
        }
        if let plist = Bundle.main.object(forInfoDictionaryKey: "REVENUECAT_API_KEY") as? String {
            let trimmed = plist.trimmingCharacters(in: .whitespacesAndNewlines)
            if !trimmed.isEmpty, trimmed != "$(REVENUECAT_API_KEY)" {
                return trimmed
            }
        }
        return nil
    }
}
